presto-spark-common/src/main/java/com/facebook/presto/spark/PrestoSparkQueryInfo.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.spark;
+
+import com.facebook.presto.execution.ExecutionFailureInfo;
+import com.facebook.presto.execution.QueryInfo;
+import com.facebook.presto.execution.QueryStats;
+import com.facebook.presto.execution.StageInfo;
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.concurrent.Immutable;
+
+import java.util.Optional;
+
+import static com.facebook.presto.server.protocol.QueryResourceUtil.globalUniqueNodes;
+import static java.util.Objects.requireNonNull;
+
+@Immutable
+public class PrestoSparkQueryInfo
+{
+    // the following fields are based on com.facebook.presto.jdbc.QueryStats,
+    // with the following fields removed:
+    //  - queued
+    //  - queuedSplits
+    //  - runningSplits
+    //  - completedSplits: duplciate of totalSplits in Presto-on-Spark
+    //  - queuedTimeMillis
+    //  - rootStage: it can be too verbose
+    private final String queryId;
+    private final String state;
+    private final int nodes;
+    private final int totalSplits;
+    private final long cpuTimeMillis;
+    private final long wallTimeMillis;
+    private final long elapsedTimeMillis;
+    private final long processedRows;
+    private final long processedBytes;
+    private final long peakMemoryBytes;
+    private final long peakTotalMemoryBytes;
+    private final long peakTaskTotalMemoryBytes;
+
+    Optional<ExecutionFailureInfo> failureInfo;
```

Comment:
optional: Maybe try to move the `ExecutionFailureInfo ` to `presto-comon`, not sure if it is fine to include presto-main in verifier.

Reply:
@arhimondr : Verifier already depends on `presto-main`:

```
https://github.com/prestodb/presto/blob/740e18c29f53a587c8398317e567cb84f2bf3f42/presto-verifier/pom.xml#L53-L56
```

Although I think it's generally a good idea to refactor common piece in different `QueryInfo/QueryStats/...` into `presto-common` (we already have too many different type of `QueryInfo/Stats` in many places. But this seems to beyond the scope of this PR 😃